StripeConnect/StripeConnect/Source/Internal/Webview/MessageHandlers/Helpers/ScriptMessageHandler.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  ScriptMessageHandler.swift
+//  StripeConnect
+//
+//  Created by Mel Ludowise on 5/1/24.
+//
+
+import WebKit
+
+/// Convenience class that conforms to WKScriptMessageHandler and can be instantiated with a closure
+class ScriptMessageHandler<T: Decodable>: NSObject, WKScriptMessageHandler {
+    let name: String
+    let didReceiveMessage: (T) -> Void
+    
+    init(name: String,
+         didReceiveMessage: @escaping (T) -> Void) {
+        self.name = name
+        self.didReceiveMessage = didReceiveMessage
+    }
+    
+    func userContentController(_ userContentController: WKUserContentController,
+                               didReceive message: WKScriptMessage) {
+        guard message.name == name else {
+            debugPrint("Unexpected message name: \(message.name)")
+            return
+        }
+        guard 
+            let bodyData = (message.body as? String)?.data(using: .utf8),
+            let body = try? JSONDecoder().decode(T.self, from: bodyData) else {
+            debugPrint("Failed to decode body for message with name: \(message.name)")
```

Comment:
Optional: You could put TODOs here for [MXMOBILE-2491](https://jira.corp.stripe.com/browse/MXMOBILE-2491) since we plan to log these errors to analytics.